android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -397,19 +397,22 @@ private void setNtpRecyclerView(LinearLayoutManager linearLayoutManager) {
         mIsBraveStatsEnabled = shouldDisplayBraveStats();
 
         if (mNtpAdapter == null) {
-            mNtpAdapter = new BraveNtpAdapter(mActivity, this, Glide.with(mActivity),
-                    mNewsItemsFeedCard, mBraveNewsController, mMvTilesContainerLayout,
-                    mNtpImageGlobal, mSponsoredTab, mWallpaper, mSponsoredLogo,
-                    mNTPBackgroundImagesBridge, false, mRecyclerView.getHeight(),
-                    mIsTopSitesEnabled, mIsBraveStatsEnabled, mIsDisplayNews, mIsDisplayNewsOptin);
-
-            mRecyclerView.setAdapter(mNtpAdapter);
-
-            if (mRecyclerView.getItemAnimator() != null) {
-                RecyclerView.ItemAnimator itemAnimator = mRecyclerView.getItemAnimator();
-                if (itemAnimator instanceof SimpleItemAnimator) {
-                    SimpleItemAnimator simpleItemAnimator = (SimpleItemAnimator) itemAnimator;
-                    simpleItemAnimator.setSupportsChangeAnimations(false);
+            if (mActivity != null && !mActivity.isDestroyed() && !mActivity.isFinishing()) {
```

Comment:
a minor suggestion, that `if`, could be combined with `if (mNtpAdapter == null)`

Reply:
If we combined with `if (mNtpAdapter == null)` then we have to change `else` with `else if(mNtpAdapter!=null)` so I think it's fine to keep as it is. wdyt?

Reply:
yes, you are right, my bad, let's keep as is.